### PR TITLE
For #147: added mutable index pattern tests

### PR DESCRIPTION
**Pull request by paulodamaso**
For #147: 
- added tests for mutable index: simple assignment and `++`, `--`, `+=`, `-=`, `*=`, `/=` and `%=` operators
- left puzzle for implementation of pattern

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#240](https://github.com/yegor256/aibolit/issues/240) is assigned to @lyriccoder[/z](https://www.0crat.com/u/lyriccoder), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/099e0643-8070-4111-9fae-f4b77812e8cf); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: test/patterns/mutable_index/test_mutable_index.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexDivisionAssignment.java')),
+            'Should match index division assignment (/=)'
+        )
+
+    @unittest.skip("Not implemented")
+    def test_find_mutable_index_mod_assignment(self):
+        self.assertEqual(
+            [4],
+            MutableIndex().value(Path(self.dir_path, 'MutableIndexModAssignment.java')),
```
Comment (lyriccoder) [inline test/patterns/mutable_index/test_mutable_index.py:71]: 1) You have added tests for such operators as `i *= 2`, but you haven't added test for cases i  = i * 1 (excepting `i = i + 1`

So, pls add tests for the following cases:
`i = i - 1`
`i = i / 2`, etc. 

Consider all operators

2) Please add test with nested `for` cycles 
```
for (i = 1; i < 5; ++i ) {
 for (j = 1; j < 5; ++j) {
      i++;
  } 
} 
```

3) add test for post-fix operator
```
i--;
i++;
```

4) Please add test with nested `for` cycles and hiding the scope
```
for (i = 1; i < 5; ++i ) {
	for (i = 1; i < 5; ++i ) {
		i  = i + 1; // what cycle is changing a variable?
	} 
} 
```


4) Please add test with `for`, `while` cycle without block {} 
```
for (i = 1; i < 5; ++i ) 
   i = i - 1 
```


5) Assigning some value to variables is also changing the variable
```
for (i = 1; i < 5; ++i ) {
   int j = 1000;
}
```

Comment (paulodamaso) [reply test/patterns/mutable_index/test_mutable_index.py:71]: @lyriccoder Some considerations:

> 1. You have added tests for such operators as `i *= 2`, but you haven't added test for cases i  = i * 1 (excepting `i = i + 1`
> 
> So, pls add tests for the following cases:
> `i = i - 1`
> `i = i / 2`, etc.
> 
> Consider all operators
The expression is not important here, the important is the assignment; we don't need to test for all operations

> 
> 1. Please add test with nested `for` cycles
> 
> ```
> for (i = 1; i < 5; ++i ) {
>  for (j = 1; j < 5; ++j) {
>       i++;
>   } 
> } 
> ```
> 
> 1. add test for post-fix operator
> 
> ```
> i--;
> i++;
> ```
> 
> 1. Please add test with nested `for` cycles and hiding the scope
> 
> ```
> for (i = 1; i < 5; ++i ) {
> 	for (i = 1; i < 5; ++i ) {
> 		i  = i + 1; // what cycle is changing a variable?
> 	} 
> } 
> ```
> 
> 1. Please add test with `for`, `while` cycle without block {}
> 
> ```
> for (i = 1; i < 5; ++i ) 
>    i = i - 1 
> ```

Tests added

> 
> 1. Assigning some value to variables is also changing the variable
> 
> ```
> for (i = 1; i < 5; ++i ) {
>    int j = 1000;
> }
> ```
Assignment is already being tested in the first case.

**Comment by paulodamaso** [comment]
@acheshkov ping

**Comment by acheshkov** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@acheshkov These errors were introduced in another commit, please take a look

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#240](https://github.com/yegor256/aibolit/issues/240) is already in scope

**Comment by 0crat** [comment]
Job audit: performer @lyriccoder[/z](https://www.0crat.com/u/lyriccoder)[/z](https://www.0crat.com/u/lyriccoder) didn't make CR comments

**Comment by 0crat** [comment]
Job `gh:yegor256/aibolit#240` is not assigned, can't get performer

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@4ea0ebc5-1e90-4d29-b5a1-8c93eef19bcd, thread: PQ-CT5UB7XAP
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.in.orders.close_job
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: java.lang.OutOfMemoryError: unable to create new native thread
java.lang.Thread[-2] java.lang.OutOfMemoryError: unable to create new native thread
```

[0.53.13](https://github.com/zerocracy/farm): CID: [285c6220-ac7d-42c8-8037-78ee4b4cfd83](https://www.0crat.com/footprint/CT5UB7XAP/285c6220-ac7d-42c8-8037-78ee4b4cfd83), Type: "Close job"

**Pull request merged**